test/cpp/end2end/xds_end2end_test.cc

```diff
@@ -2390,6 +2390,29 @@ TEST_P(LdsRdsTest, RouteMatchHasUnsupportedSpecifier) {
             "No prefix field found in Default RouteMatch.");
 }
 
+// Tests that LDS client should send a NACK if route match has a case_sensitive
+// set to false.
+TEST_P(LdsRdsTest, RouteMatchHasCaseSensitiveFalse) {
+  gpr_setenv("GRPC_XDS_EXPERIMENTAL_ROUTING", "true");
```

Comment:
Please also add a test that shows that if xds_routing is not enabled, case_sensitive is ignored (instead of causing the config to be rejected).